csvqb/csvqb/models/cube/csvqb/components/attribute.py

```diff
@@ -20,10 +33,17 @@ class QbAttribute(ColumnarQbDataStructureDefinition, ABC):
     def is_required(self) -> bool:
         pass
 
+    @abstractmethod
+    def new_attribute_values(self) -> List[NewQbAttributeValue]:
+        pass
+
 
 @dataclass
 class ExistingQbAttribute(QbAttribute):
     attribute_uri: str
+    new_attribute_values: List[NewQbAttributeValue] = field(
```

Comment:
@CharlesRendle just an FYI that I decided that `ExistingQbAttribute`s should be able to define new_attribute_values as well.

Reply:
I'm not sure whether it's worth having a `from_data` method here though, since this isn't the easy path for a user. If they're doing something as exotic as creating `NewQbAttributeValue`s whilst using an `ExistingQbAttribute`, I think they can write their own code to figure out what those `NewQbAttributeValue`s are.

Reply:
Agree. Defining `NewQbAttributeValue` using an `ExistingQbAttribute` cannot be done with an airgapped computer, so surfacing this functionality means "Congrulations, your library's scope is now 400% larger."